sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/tests/EventHubEndToEndTests.cs

```diff
@@ -209,6 +210,75 @@ public async Task EventHub_PartitionKey()
             }
         }
 
+        [Test]
+        public async Task EventHub_InitialOffsetFromStart()
+        {
+            using (var host = BuildHost<EventHubTestSendOnlyJobs>().Item1)
+            {
+                var method = typeof(EventHubTestSendOnlyJobs).GetMethod(nameof(EventHubTestSendOnlyJobs.SendEvent_TestHub), BindingFlags.Static | BindingFlags.Public);
+                await host.CallAsync(method, new { input = _testId });
+            }
+            var initialOffsetOptions = new InitialOffsetOptions()
+            {
+                Type = "FromStart"
+            };
+            using (var host = BuildHost<EventHubTestInitialOffsetFromEndJobs>(initialOffsetOptions: initialOffsetOptions).Item1)
+            {
+                bool result = _eventWait.WaitOne(Timeout);
+                Assert.True(result);
+            }
+        }
+
+        [Test]
+        public async Task EventHub_InitialOffsetFromEnd()
+        {
+            // Send a message to ensure the stream is not empty as we are trying to validate that no messages are delivered in this case
+            using (var host = BuildHost<EventHubTestSendOnlyJobs>().Item1)
+            {
+                var method = typeof(EventHubTestSendOnlyJobs).GetMethod(nameof(EventHubTestSendOnlyJobs.SendEvent_TestHub), BindingFlags.Static | BindingFlags.Public);
+                await host.CallAsync(method, new { input = _testId });
+            }
+            var initialOffsetOptions = new InitialOffsetOptions()
+            {
+                Type = "FromEnd"
+            };
+            using (var host = BuildHost<EventHubTestInitialOffsetFromEndJobs>(initialOffsetOptions: initialOffsetOptions).Item1)
+            {
+                // We don't expect to get signalled as there should be no messages received with a FromEnd initial offset
+                bool result = _eventWait.WaitOne(Timeout);
```

Comment:
Are we waiting 30 seconds to verify this?

Reply:
Can we start a host and post a single message using a client and make sure only that one is received?

Reply:
Yes, the Timeout is 30s.